File: app/src/main/java/pl/mosenko/sunnypodlaskie/persistence/dao/AbstractBaseDaoImpl.java
package pl.mosenko.sunnypodlaskie.persistence.dao;

import com.j256.ormlite.support.ConnectionSource;
import com.j256.ormlite.table.DatabaseTableConfig;

import java.sql.SQLException;

import pl.mosenko.sunnypodlaskie.persistence.entities.BaseOrmLiteEntity;

/**
 * Created by syk on 18.05.17.
 */

public abstract class AbstractBaseDaoImpl<DataType extends BaseOrmLiteEntity, IdType> extends RxBaseDaoImpl<DataType, Long> implements OrmLiteEnityDAO<DataType> {

    protected AbstractBaseDaoImpl(Class<DataType> dataClass) throws SQLException {
        super(dataClass);
    }

    protected AbstractBaseDaoImpl(ConnectionSource connectionSource, Class<DataType> dataClass) throws SQLException {
        super(connectionSource, dataClass);
    }

    protected AbstractBaseDaoImpl(ConnectionSource connectionSource, DatabaseTableConfig<DataType> tableConfig) throws SQLException {
        super(connectionSource, tableConfig);
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/persistence/dao/AbstractBaseDaoImpl.kt
package pl.mosenko.sunnypodlaskie.persistence.dao

import com.j256.ormlite.support.ConnectionSource
import com.j256.ormlite.table.DatabaseTableConfig
import pl.mosenko.sunnypodlaskie.persistence.entities.BaseOrmLiteEntity

/**
 * Created by syk on 18.05.17.
 */
abstract class AbstractBaseDaoImpl<DataType : BaseOrmLiteEntity?, IdType> : RxBaseDaoImpl<DataType?, Long?>, OrmLiteEnityDAO<DataType?> {
    protected constructor(dataClass: Class<DataType?>?) : super(dataClass) {}
    protected constructor(connectionSource: ConnectionSource?, dataClass: Class<DataType?>?) : super(connectionSource, dataClass) {}
    protected constructor(connectionSource: ConnectionSource?, tableConfig: DatabaseTableConfig<DataType?>?) : super(connectionSource, tableConfig) {}
}
